splipy/basis_eval.pyx

```diff
@@ -0,0 +1,122 @@
+from bisect import bisect_right, bisect_left
+import numpy as np
+cimport numpy as np
+import copy
+cimport cython
+from cpython cimport bool
+@cython.boundscheck(False) # turn off bounds-checking for entire function
+
+cdef inline int int_max(int a, int b): return a if a >= b else b
+cdef inline unsigned int uint_min(unsigned int a, unsigned int b): return a if a <= b else b
+cdef inline double fabs(double a):     return a if a > 0 else -a
```

Comment:
I'm not sure these are very useful TBH.

Reply:
They come from [this cython tutorial](http://cython.readthedocs.io/en/latest/src/userguide/numpy_tutorial.html#adding-types)

Quote:

> The builtin min and max functions works with Python objects, and are
> so very slow. So we create our own.
>  "cdef" declares a function which has much less overhead than a normal
>  def function (but it is not Python-callable)
>  "inline" is passed on to the C compiler which may inline the functions
>  The C type "int" is chosen as return type and argument types
>  Cython allows some newer Python constructs like "a if x else b", but
>  the resulting C file compiles with Python 2.3 through to Python 3.0 beta.
> `cdef inline int int_max(int a, int b): return a if a >= b else b`
> `cdef inline int int_min(int a, int b): return a if a <= b else b`

The question is if these functions are called enough to actually make a difference. From my limited experimentation, then I would have to guess that "no, they are not".

Reply:
I was just concerned about whether the C-compiler actually decides to inline them, because in the C-file they don't look so simple.